### PR TITLE
Limit checkout to registered and logged in users

### PR DESCRIPTION
**Pull request by grwthomps**

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -18,15 +18,20 @@ def create
           price: item.price
           })
       end
+      order.item_orders.each do |item_order|
+        item = Item.find(item_order.item_id)
+        item.reduce_quantity(item_order.quantity)
+        item.save
+      end
```
Comment (johnktravers) [inline app/controllers/orders_controller.rb:25]: There's probably a better way of reducing item quantity. We can ask Meg at check ins.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -17,12 +17,47 @@ def create
...
+      elsif params[:password] != params[:password_confirmation]
+        flash.now[:error] = ['Password confirmation doesn\'t match Password']
+        render :edit
+      elsif user.update(user_params)
+        flash[:success] = ['You have successfully updated your password!']
+        redirect_to '/profile'
+      else
+        flash.now[:error] = user.errors.full_messages
+        render :edit
+      end
```
Comment (johnktravers) [inline app/controllers/users_controller.rb:57]: Let's write some helper methods for these conditions

File: spec/features/default_user/profile_spec.rb [hunk 3]
```diff
@@ -0,0 +1,100 @@
+require 'rails_helper'
+
+RSpec.describe 'As a default user' do
```
Comment (johnktravers) [inline spec/features/default_user/profile_spec.rb:3]: We need to make this test for the merchant and admin user types

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (zacisaacson) [inline app/controllers/users_controller.rb:57]: Looks great

**Review by zacisaacson — APPROVED** [review]

**Review by johnktravers — COMMENTED** [review]
We need to write a few more tests for different user types, but other than that everything looks great!

**Pull request merged**